mbed_greentea/mbed_report_api.py

```diff
@@ -165,11 +166,11 @@ def exporter_testcase_junit(test_result_ext, test_suite_properties=None):
         for test_suite_name in test_results:
             test = test_results[test_suite_name]
 
-            # tc_elapsed_sec = test['elapsed_time']
-            tc_stdout = str() #test['single_test_output']
-
             try:
-                tc_stdout = test['single_test_output'].decode('unicode_escape').encode('ascii','ignore')
+                if sys.version_info >= (3, 0):
+                    tc_stdout = str(test['single_test_output']).encode('ascii', 'ignore')
+                else:
+                    tc_stdout = test['single_test_output'].decode('unicode_escape').encode('ascii', 'ignore')
```

Comment:
Was the previous implementation not working in Python 3? Or does this provide better output?

Reply:
It was failing and I couldn't figure out a decent way of making it work for both python versions, I suppose I could revisit it.

test['single_test_output'] is a byte array in python2 and a string in python3 (which doesn't support encoding/decoding

Reply:
AH, well thanks for fixing it then 👍